src/ContentDecorator.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Widgets;
+
+use Yiisoft\View\Exception\ViewNotFoundException;
+use Yiisoft\View\WebView;
+use Yiisoft\Widget\Widget;
+
+/**
+ * ContentDecorator records all output between {@see begin()} and {@see end()]} calls, passes it to the given view file
+ * as `$content` and then echoes rendering result.
+ *
+ * ```php
+ * <?php ContentDecorator::begin()
+ *     ->viewFile('@app/views/layouts/base.php'),
+ *     ->params([]),
+ *     ->view($this),
+ * ]) ?>
+ *
+ * some content here
+ *
+ * <?php ContentDecorator::end() ?>
+ * ```
+ */
+class ContentDecorator extends Widget
```

Comment:
Could be final?

Reply:
Because they are base widgets that could be extended.